### ISSUE TITLE
Terraform Test - fails to init when test name has spaces

### ISSUE THREAD

**Issue by myerscf** [hashicorp/terraform#34372]
### Terraform Version

```shell
1.6.4
```

### Terraform Configuration Files

```terraform
...terraform config...
```

### Debug Output

Error: Module not installed
on s3.tftest.hcl line 35, in run "it makes s3":

### Expected Behavior

Tests can have names with spaces, so should be able to init modules for testing.

### Actual Behavior

Cannot init modules until spaces are removed from test name

### Steps to Reproduce

```hcl
run "it makes s3" {
  command = plan
  module {
    source = "./modules/s3"
  }
}
```

**Comment by apparentlymart** [comment]
Thanks for reporting this, @myerscf!

It seems that the bug here is that the test language decoder didn't check that the test name is a valid identifier, and then some other downstream code failed in a strange way because it was expecting all test runs to have valid names.

Therefore I expect that we will fix this by introducing a better error message for this case, explaining what the rules are for test run names. (The rules here will be the same as the rules for what makes a valid variable name or resource name in the main Terraform language; in particular, spaces are not allowed.)

**Comment by liamcervante** [comment]
Just to reiterate Martin's comment, it's not valid to include spaces in a run block name as these are HCL identifiers (since you can refer back to previous run blocks).

The PR I've merged ensures a targeted error message is produced when the configuration is parsed if the run block names are invalid. This will be released in the 1.7.x series.
